HPCC-16147 Fix keyedjoin skip bug.

jakesmith opened the pull request:
A SKIP in a keyed join transform would skip all RHS matches in
the right group.

Signed-off-by: Jake Smith jake.smith@lexisnexis.com

wangkx committed:
HPCC-16036 Add option to bypass cached WUResult in WsWorkunits

WUResult may be cached for better performance. When a workunit
is resubmitted, new WUResult may not be retrieved before old
WUResult cache is timed out. In this fix, an option is added to
bypass the cached WUResult.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

jakesmith committed:
HPCC-15895 Roxie NWay expanded input v input stream handling fixes

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

ghalliday committed:
HPCC-15845 Add support for average/sd on query stats

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

Michael-Gardner committed:
HPCC-15142 Minimal changes needed for DESTDIR

Workflow:
export DESTDIR=$HOME/testdir
cmake -DRUNTIME_USER=$USER ../HPCC-Platform/
make install
make configure
$DESTDIR/etc/init.d/hpcc-init start

Signed-off-by: Michael Gardner <michael.gardner@lexisnexis.com>

ghalliday committed:
HPCC-15925 Minor code generator speed ups

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
HPCC-15925 Reduce the overhead of annotations by 8 bytes

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

AttilaVamos committed:
HPCC-15922 Despray to a single Landing Zone fails

Store computer name and related drop zone number as propety value.

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

ghalliday committed:
HPCC-15925 Remove mutex if hql compiled as single threaded

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
HPCC-15925 Remove extra 8 bytes from annotations

Also reduces the redundant processing of infoFlags in annotations

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

Commit:
HPCC-15636 envgen - set thor master option

- set_ips thor <IPs> will use first IP as thor master

Signed-off-by: Gleb Aronsky <gleb.aronsky@lexisnexis.com>

Commit:
HPCC-15962 Envgen - add options to specify multiple esp instances

- Add flag -espnodes to specify number of esp nodes wizard should generate

Signed-off-by: Gleb Aronsky <gleb.aronsky@lexisnexis.com>

wangkx committed:
HPCC-15481 Add 2 transaction ID functions to logging

Also revised the code to avoid duplicated code in CCassandraLogAgent
and CMySQLLogAgent:
1. Moved ensureInputString(), CLogTable::loadMappings(), and
CLogGroup::loadMappings() to datafieldmap.cpp;
2. Moved the code for reading LogGroup Cfg to readLogGroupCfg();
Moved the code for reading LogSource Cfg to readLogSourceCfg();
3. Created a base class CDBLogAgentBase for CCassandraLogAgent
and CMySQLLogAgent. Moved getTransactionSeed(), updateLog(),
checkLogSource(), buildUpdateLogStatement(), appendFieldInfo(),
addMissingFields(), getLoggingTransactionID(), and filterLog
Content() from CCassandraLogAgent to CDBLogAgentBase. Those
methods can be shared with CMySQLLogAgent.
4. Moved the code for reading DB Cfg to readDBCfg(); Moved the
code for reading Transaction Cfg to readTransactionCfg().

Also: 1. for getTransID(), read method name from trans data array;
2. in cassandra logging agent, set DB server/userID/password every
time before creating a new connection; 3. Change 3 #define's to
static; 4. throw exception if no init() called; 5. add createLocal
TransactionSeed() used for local Transaction ID.

Shared files is moved into logginglib

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

ghalliday committed:
HPCC-15995 Implementation of scanning heaplets

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
HPCC-16018 Implement blocked heap allocators

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

rpastrana committed:
HPCC-15612 Libcouchbase as an embedded language in ECL

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

ghalliday committed:
HPCC-15995 Allow heapflags to be configured in thor

Mainly here to allow testing of different allocators, but also in
preparation for thor to use different flags depending on the context.

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

Michael-Gardner committed:
HPCC-15142 Documentation Changes for make install

* Root level CMakeLists.txt documentation changes for building
   a development installation
* Modifications to initfiles/etc/DIR_NAME/CMakeLists.txt for
   cleanup of unnecessary comments

Signed-off-by: Michael Gardner <michael.gardner@lexisnexis.com>

rpastrana committed:
HPCC-15612 Code Review changes

- Searches for libcouchbase in externals folder
- Splits out class declarations and implementations
- Declares several static methods

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

Michael-Gardner committed:
Change to PR - Michael Gardner

Signed-off-by: Michael Gardner <michael.gardner@lexisnexis.com>

rpastrana committed:
HPCC-15612 Merging M. Gardner's CMAKE changes

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

Michael-Gardner committed:
HPCC-15142 DESTDIR moved to cmake variable

Signed-off-by: Michael Gardner <michael.gardner@lexisnexis.com>

Michael-Gardner committed:
HPCC-15142 OSX compatibility changes

Signed-off-by: Michael Gardner <michael.gardner@lexisnexis.com>

ghalliday committed:
Merge pull request #8876 from shamser/issue15751

HPCC-15751 Use getLikelihood of filter cond to decide whether to spill

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge pull request #8807 from ghalliday/issue15767

HPCC-15767 Refactor some of the volatile code in preparation for HPCC-2946

Reviewed-By: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
HPCC-15925 Changes following review

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

rpastrana committed:
HPCC-15612 Target hpcc-systems' fork of libcouchbase-cxx

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

richardkchapman committed:
Merge pull request #8986 from dcamper/hpcc-16046-kafka-valgrind

HPCC-16046 Avoid potential invalid memory access or garbage output in Kafka plugin

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8923 from ghalliday/issue15965

HPCC-15965 Reorder inheritance for various significant classes

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

g-pan committed:
HPCC-16073 DOCS: Fix Errors generating docs

FIX HPCC-16073 DOCS: Fix Errors generating docs
Fixing Errors causing build to fail.

Signed-off-by: G Panagiotatos <greg.panagiotatos@lexisnexis.com>

RussWhitehead committed:
HPCC-15864 Investigate improvements to LDAP cache

In a previous code review, the following recommendations were made for
permissions cache improvements.

Use a map to a CachedUser instead of a CachedUser *
Avoid some appending to temporary buffers
Examine why clone() is called on the user instead of LINK()

Although the CachedUser is already stored in a map, the other two
bullets are areas that needed improvement. This PR removes
numerous unnecessary intermediate buffers, and LINKs the
cached_user instead of cloning it

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

richardkchapman committed:
Merge pull request #8989 from g-pan/H16073-FixErr

HPCC-16073 DOCS: Fix Errors generating docs

Reviewed-By: Jim DeFabia <james.defabia@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.x' into candidate-6.0.4

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Community Edition 6.0.4-1 Gold

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.4' into candidate-6.0.6

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.6'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

g-pan committed:
HPCC-16076 DOCS: Fix Errors generating docs

FIX HPCC-16076 DOCS: Fix Errors generating docs
Fixing Errors causing build to fail.

Signed-off-by: G Panagiotatos <greg.panagiotatos@lexisnexis.com>

richardkchapman committed:
Merge pull request #8991 from g-pan/H16076-6041BuildErr

HPCC-16076 DOCS: Fix Errors generating docs

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8990 from AttilaVamos/HPCC-15922-fix-6.0.6

HPCC-15922 Despray to a single Landing Zone fails

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8948 from ghalliday/issue16009

HPCC-16009 Improve the code for CRowStreamReader with no limit

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.x' into candidate-6.0.4

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Community Edition 6.0.4-1 Gold (retag)

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.4' into candidate-6.0.6

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.6'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #8981 from jakesmith/hpcc-16004

HPCC-16004 UPDATE regression fix

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
HPCC-8742 Changes following review

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
HPCC-15933 Fix mismerge from 6.0.4 to master

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

richardkchapman committed:
Merge pull request #8993 from ghalliday/issue15933b

HPCC-15933 Fix mismerge from 6.0.4 to master

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
HPCC-16011 Changes following review

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

richardkchapman committed:
Merge pull request #8940 from ghalliday/issue8742

HPCC-8742 Add a dfsls command to daliadmin/daliadmin

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8942 from ghalliday/issue16011

HPCC-16011 Add a daliadmin dfscheck command to locate dfs problems

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8930 from wangkx/h15481

HPCC-15481 Add 2 transaction ID functions to logging

Reviewed-By: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

rpastrana committed:
HPCC-16083 Interface enhancements to help UI development

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

ghalliday committed:
Merge pull request #8988 from RussWhitehead/caching

HPCC-15864 Investigate improvements to LDAP cache

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
HPCC-16088 Excessive calls to install_name_tool

Also suppress the errors from calling it on script files.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-16090 - Build error on Clang in new logging code

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #8996 from richardkchapman/build-error-logging

HPCC-16090 - Build error on Clang in new logging code

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

jakesmith committed:
HPCC-15978 Possible deadlock on blocking splitter when stopping

Avoid calling prepareInput() from writer thread, which is only
started after an input has already called prepareInput()
By avoiding calling it, it avoid the potential deadlock, because
the writer thread could be being stopped by the laster output
stopping, in which case it will be blocking a CS whilst waiting
for it's thread to complete. Whilst the writer thread could be
blocked by the entry into the unecessary call to prepareInput()

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

richardkchapman committed:
HPCC-16085 MySQL plugin connection cache needs to retry on failure

When reusing a cached MySQL connection, we need to retry on failures in case
the MySQL server has restarted.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #8997 from richardkchapman/mysql-cache-stale

HPCC-16085 MySQL plugin connection cache needs to retry on failure

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
HPCC-16084 MySQL Plugin connection cache - retire old connections

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

jakesmith committed:
HPCC-16092 Log filename CDelayedDelete is pending on.

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

afishbeck committed:
HPCC-16060 Fix regression causing roxie log of query id to be lost

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

ghalliday committed:
Merge pull request #8998 from richardkchapman/mysql-cache-timeout

HPCC-16084 MySQL Plugin connection cache - retire old connections

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
HPCC-15845 Fix a couple of issues following own review

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

richardkchapman committed:
HPCC-16091 MySQL Plugin connection cache - make values configurable

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #9002 from richardkchapman/mysql-cache-config

HPCC-16091 MySQL Plugin connection cache - make values configurable

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge pull request #9000 from afishbeck/roxieLogQueryId

HPCC-16060 Fix regression causing roxie log of query id to be lost

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8999 from jakesmith/hpcc-16092

HPCC-16092 Log filename CDelayedDelete is pending on.

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #9001 from jakesmith/hpcc-15978

HPCC-15978 Possible deadlock on blocking splitter when stopping

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge pull request #8995 from richardkchapman/install_name_tool

HPCC-16088 Excessive calls to install_name_tool

Reviewed-By: Michael Gardner <michael.gardner@lexisnexis.com>

jakesmith committed:
HPCC-16093 Fix potential deadlock race condition

If during funnel stop, a input handler triggers full condition
before the stop checks it, then deadlock.

Also fix minor row leak.

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

Michael-Gardner committed:
HPCC-16098 Cmake configure target now sets up ssh keys if not present

Signed-off-by: Michael Gardner <michael.gardner@lexisnexis.com>

GordonSmith committed:
HPCC-16061 Improve nest datasets in result viewer

Instead of nesting HTML grids (with headers per row), use compound columns and vertically pad cells as needed.

Fixes HPCC-16061

Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

richardkchapman committed:
Merge pull request #9003 from jakesmith/hpcc-16093

HPCC-16093 Fix potential deadlock race condition

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8992 from jakesmith/hpcc-15895

HPCC-15895 Roxie NWay expanded input v input stream handling fixes

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8956 from Michael-Gardner/hpcc-15142-2

HPCC-15142 Minimal changes needed for 'make install'

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8904 from ghalliday/issue15845

HPCC-15845 Add support for average/sd on query stats

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8984 from JamesDeFabia/16059ImproveCodeExample

HPCC-16059 Improve example code for STD.Str.Contains

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

GordonSmith committed:
HPCC-16096 Add "legacy" mode for results.

Fixes HPCC-16096

Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

richardkchapman committed:
Merge pull request #8931 from rpastrana/HPCC-15612-couchbaseplugin

Hpcc 15612 Embedded Couchbase Plugin

Reviewed-By: Jamie Noss <james.noss@lexisnexis.com>
Reviewed-By: Dan Camper <dan.camper@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

Michael-Gardner committed:
HPCC-16099 Add check to init-functions for getopt compatibility

Signed-off-by: Michael Gardner <michael.gardner@lexisnexis.com>

rpastrana committed:
HPCC-16083 Add new List ESP bindings of type desdl method

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

RussWhitehead committed:
HPCC-16086 Security permissions values are hardcoded

The code for Dali, workunit and plugins etc includes hardcoded values for
permission settings. Instead of using mysterious values like 255, -255, -1
and 0, we should specify the named SecAccess_* enumerations found in
seclib.hpp

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

richardkchapman committed:
Merge pull request #9005 from GordonSmith/HPCC-16096

HPCC-16096 Add "legacy" mode for results.

Reviewed-By: Miguel Vazquez <miguel.vazquez@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-16097 Memory leak when using Java plugin

Java allocations made when calling iterator for dataset results
would not be released util thread terminated.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

jakesmith committed:
HPCC-16051 Fix problems with child act. initialization

Activities with variant initialization, e.g. dynamic filenames
were not being initialized correctly if in loop or child queries.
In some cases it could cause a crash (this JIRA was an incident of
that), in others it would fail to re-initialize based on the
context. Also changed so that only the child activities that
need initialization, request it, which in many cases means none.

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

GordonSmith committed:
HPCC-15857 GraphTreeWidget missing tooltips

Fixes HPCC-15857

Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

Michael-Gardner committed:
Cmake required version updated to 3.3.2

Signed-off-by: Michael Gardner <michael.gardner@lexisnexis.com>
Signed-off-by: Cloud User <centos@cent-7-sel.novalocal>

wangkx committed:
HPCC-16089 Fix trigraph warning in esp logging code

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

rpastrana committed:
HPCC-16083 Code review changes

- Addresses formating issues
- Addresses nil entries in XML requests
- Fixes versioning
- Makes SDS_LOCK_TIMEOUT unique name: SDS_LOCK_TIMEOUT_DESDL


Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

richardkchapman committed:
HPCC-16106 eclcc cores when using -qa and -main together

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

GordonSmith committed:
HPCC-13695 Simplify error messages "copy to clipboard"

Enable copy/download toolbar

Fixes HPCC-13695

Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

rpastrana committed:
HPCC-16083 Addressed issues from 2nd code review

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

richardkchapman committed:
Merge pull request #9014 from richardkchapman/java-argleak

HPCC-16097 Memory leak when using Java plugin

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>

richardkchapman committed:
Merge pull request #9008 from Michael-Gardner/hpcc-16099

HPCC-16099 Add check to init-functions for getopt compatibility

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9015 from richardkchapman/eclcc-qa-core

HPCC-16106 eclcc cores when using -qa and -main together

Reviewed-By: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>

richardkchapman committed:
Merge pull request #9010 from RussWhitehead/SecEnum

HPCC-16086 Security permissions values are hardcoded

Reviewed-By: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8976 from wangkx/h15659

HPCC-16036 Add option to bypass cached WUResult in WsWorkunits

Reviewed-By: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8983 from ghalliday/issue16041

HPCC-16041 Indicate if inline tables can be filtered

Reviewed-By: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>
Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9009 from rpastrana/HPCC-16083

Hpcc 16083 Provide ESP Binding information from WsESDLConfig

Reviewed-By: Kevin Wang <kevin.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9011 from GordonSmith/HPCC-15857

HPCC-15857 GraphTreeWidget missing tooltips

Reviewed-By: Miguel Vazquez <miguel.vazquez@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Community Edition 6.0.6-rc1 Release Candidate 1

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

rpastrana committed:
HPCC-16109 WsEsdlConfig OSX build break

- Addresses build break in osx due to SMCLib not explicitly included

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

richardkchapman committed:
Merge pull request #9017 from rpastrana/HPCC-16109-OSXBuildBreak

HPCC-16109 WsEsdlConfig OSX build break

Reviewed-By: Kevin Wang <kevin.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

wangkx committed:
HPCC-16089 Modify based on code review

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

rpastrana committed:
HPCC-16108 Add UI centric enhancements

- GetESDLBindings can provide list of methods available
- GetESDLDefinition can provide list of methods available

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

rpastrana committed:
HPCC-16108 Several changes based off code review

- Removes \t
- Declares request of nil_remove type
- utilizes existing var

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

Commit:
HPCC-16081 Improve documentation for Dali setting filesDefaultUser

Signed-off-by: Jim DeFabia <jamesdefabia@lexisnexis.com>

richardkchapman committed:
Merge pull request #9016 from GordonSmith/HPCC-13695

HPCC-13695 Simplify error messages "copy to clipboard"

Reviewed-By: Miguel Vazquez <miguel.vazquez@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9019 from rpastrana/HPCC-16108-MoreUICentricChanges

HPCC-16108 Add UI centric enhancements

Reviewed-By: Kevin Wang <kevin.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9012 from wangkx/h16089

HPCC-16089 Fix trigraph warning in esp logging code

Reviewed-By: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-16116 - Improve tracing when setting thread priorities

Log the thread id not the process id.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9021 from richardkchapman/gettid

HPCC-16116 - Improve tracing when setting thread priorities

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>

jakesmith committed:
HPCC-14655 Handle environment connection reload failure

Re-establish the environment connection if it's unable to reload
on a clearCache call. reload can fail if the Environment has
been renewed and the connection to the old node has been
invalidated.

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

richardkchapman committed:
HPCC-16117 UDP messaging threads should run at elevated priority

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9022 from jakesmith/hpcc-14655

HPCC-14655 Handle environment connection reload failure

Reviewed-By: Attila Vamos <attila.vamos@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

Michael-Gardner committed:
Conditional changes to cmake version

Signed-off-by: Michael Gardner <michael.gardner@lexisnexis.com>

AttilaVamos committed:
HPCC-15744 Allow to '~file::<ip_address>::<filepath>' opens an external
file with any character that's invalid to a logical file path.

Separate external file and logical file handling.

Add unit test

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

rpastrana committed:
HPCC-14563 Fetch trxid from logging manager if available

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

Conflicts:

	esp/logging/logginglib/loggingagentbase.hpp

Michael-Gardner committed:
HPCC-16120 Fix type causing regression in init_thor script

The section of code for kill_slaves relies on the uslaves file
being present.  Yet in the frunssh call we use the slaves file
for our ip list, unlike every other frunssh call where we rely on
the uslaves file.  This was causing issues where the uslaves file
would be present, but a previous failure to connect to the slave
machines would erase the slaves file (as was intended), and cause
the startup to always fail.

The fix is to use the uslaves file, as was originally intended.
If there is a failure to connect to the slave machines to stop the
slaves, we want the uslaves file to remain in the instancedir so
we can try again on the next startup attempt.

Signed-off-by: Michael Gardner <michael.gardner@lexisnexis.com>

Commit:
HPCC-16057 Requesting thorslave log gives error

The thorprocess was incorrect in the drop down where download slave log is inititated. Changed response values to modify drop down.

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

richardkchapman committed:
Merge pull request #9029 from miguelvazq/HPCC-16057

HPCC-16057 Requesting thorslave log gives error

Reviewed-By: Gordon Smith <gordon.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9026 from AttilaVamos/HPCC-15744-fix-6.0.6

HPCC-15744 Allow to '~file::<ip_address>::<filepath>' opens an external file with any character that's invalid to a logical file path.

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9025 from rpastrana/HPCC-14563-newnewTrxid

HPCC-14563 Fetch trxid from logging manager if available

Reviewed-By: Kevin Wang <kevin.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-15970 Implement getDaliServers() in roxie

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

mckellyln committed:
HPCC-15945 Thor slave hang in countproject

If a slave countproject calls stop() before nextRow()
(ie if one slave hits CHOSEN limit others will be told to stop)
a deadlock could occur.  This change moves PARENT::stop() call to
after check for first in stop() so that signalNext() can signal
the semaphore inInputFinished() is waiting for.  Thus PARENT::stop()
does not deadlock waiting for onInputFinished to end.

Signed-off-by: Mark Kelly <mark.kelly@lexisnexis.com>

richardkchapman committed:
Merge pull request #9020 from JamesDeFabia/16081DOCS_filesDefaultUser

HPCC-16081 Improve documentation for Dali setting filesDefaultUser

Reviewed-By: Russ Whitehead <william.whitehead@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

rpastrana committed:
HPCC-16123 Populate return fields

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

Michael-Gardner committed:
HPCC-16124 GPG Signing changes

* GPG Signing on by default
* Check for ability to sign if SIGN_MODULES=ON
* Add public key to ecl/regress for developers to use in testing

Signed-off-by: Michael Gardner <michael.gardner@lexisnexis.com>

rpastrana committed:
HPCC-16123 Code review changes

- Versions interface changes
- Removes extra blank line

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

richardkchapman committed:
Merge pull request #9028 from Michael-Gardner/HPCC-16120

HPCC-16120 Fix typo causing regression in init_thor script

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9032 from rpastrana/HPCC-16123-configuremethod-multiple

HPCC-16123 Populate return fields

Reviewed-By: Russ Whitehead <william.whitehead@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9030 from richardkchapman/getdaliservers

HPCC-15970 Implement getDaliServers() in roxie

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>

jakesmith committed:
HPCC-16064 Log stalled splitter outputs

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

Michael-Gardner committed:
HPCC-16124 PLUGINS ON will negate SIGN_MODULES

Signed-off-by: Michael Gardner <michael.gardner@lexisnexis.com>

Michael-Gardner committed:
HPCC-16124 Swap Default for SIGN_MODULES back to off

Signed-off-by: Michael Gardner <michael.gardner@lexisnexis.com>

richardkchapman committed:
HPCC-16130 Change default timeout on ok_to_send response

If packets get lost in the ok_to_send conversation, then waiting 5 seconds
before resending is too long if the entire master->slave conversation is going
to time out in 2 seconds (as it does by default...)

Seems more sensible to default to timing out the ok_to_send in slightly less
than the master-slave.

Also, change the timeout to be specified in milliseconds rather than seconds
for consistency and granularity.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9013 from Michael-Gardner/cmake-required-version-update

Cmake required version updated to 3.3.2

Reviewed-By: Xiaoming Wang <xiaoming.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

jakesmith committed:
HPCC-16121 Problems with virtual logicalfilename in superfiles

If a superfile was being read on a cluster that was larger than
the width of the subfiles it was composed of, the wrong subfile
names were being returned for virtual logicalfilename fields.

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

richardkchapman committed:
Merge pull request #9031 from mckellyln/hpcc-15945b

HPCC-15945 Thor slave hang in countproject

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

Michael-Gardner committed:
HPCC-16103 Removed logging from common/remote/rmtssh.cpp

Signed-off-by: Michael Gardner <michael.gardner@lexisnexis.com>

mckellyln committed:
HPCC-16016 CheckAbnormalTermination could incorrectly report job failed

Signed-off-by: Mark Kelly <mark.kelly@lexisnexis.com>

rpastrana committed:
HPCC-16158 Delete methods should acknowledge optional params

- Acknowledges optional params

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

afishbeck committed:
HPCC-14207 Make sure graphids returned by WUQueryDetails are unique

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

richardkchapman committed:
Merge pull request #9038 from rpastrana/HPCC-16158-ondelete-optionals

HPCC-16158 Delete methods should acknowledge optional params

Reviewed-By: Kevin Wang <kevin.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9018 from Michael-Gardner/HPCC-16103

HPCC-16103 Removed logging from common/remote/rmtssh.cpp

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9039 from afishbeck/queryDetailsDedupeGraph14207

HPCC-14207 Make sure graphids returned by WUQueryDetails are unique

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8894 from ghalliday/issue15925

HPCC-15925 Minor code generator speed ups and memory reduction

Reviewed-By: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9004 from GordonSmith/HPCC-16061

HPCC-16061 Improve nest datasets in result viewer

Reviewed-By: Miguel Vazquez <miguel.vazquez@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.6'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

jakesmith committed:
HPCC-10132 Implement constant rhs for lookup/smart join

If the lookup join is in a loop or child query and the rhs of
the lookup is static, avoid rebuilding the rhs.

Also fixed a leak spotted in review (not a new leak) of lhs
stream object when using a distributor.

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

jakesmith committed:
HPCC-16172 Fix some usability issues with daliadmin add.

Used to silently reject xpath with a trailing '/'
Made value optional.
And allowed arbitrary xpaths to be provided, which will create
the intermediate missing nodes as well as the adding the tail.

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

mckellyln committed:
HPCC-16175 Thor issue13081.ecl multislave stable

Signed-off-by: Mark Kelly <mark.kelly@lexisnexis.com>

Commit:
HPCC-15636 envgen - set thor master option

- set_ips thor <IPs> will use first IP as thor master

Signed-off-by: Gleb Aronsky <gleb.aronsky@lexisnexis.com>

rpastrana committed:
HPCC-16162 Parse out ESDLService name for UI consumption

- Parses out, and reports ESDSLService in getESDLDefinition response
- Parses out, and reports ESDSLService in listDESDLEspBindings response

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

rpastrana committed:
HPCC-16181 Mechanism for pass-through method configuration

- DESDL Engine appends dynamic method configuration to roxie req

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

richardkchapman committed:
Merge pull request #9042 from jakesmith/hpcc-16172

HPCC-16172 Fix some usability issues with daliadmin add.

Reviewed-By: Attila Vamos <attila.vamos@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

GordonSmith committed:
HPCC-15848 Graph properties fail to display

Ensure table + header exists for helpers even when no properties exist.

Fixes HPCC-15848

Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

ghalliday committed:
HPCC-16142 Fix problems creating spills from dictionaries

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

jakesmith committed:
HPCC-15763 Increase default physical parallel delete limit.

Make default the number of parts.

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

Commit:
HPCC-16194 SocketEndpointv4 added getUrlStr(StringBuffer &) to struct

Signed-off-by: Michael Gardner <michael.gardner@lexisnexis.com>

Commit:
HPCC-16191 Add missing alerts in ECL Watch

Fixes HPCC-16191

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

richardkchapman committed:
Merge pull request #9045 from mckellyln/thor_regress

HPCC-16175 Thor issue13081.ecl multislave stable

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9051 from jakesmith/hpcc-15763

HPCC-15763 Increase default physical parallel delete limit.

Reviewed-By: Attila Vamos <attila.vamos@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9033 from jakesmith/hpcc-16064

HPCC-16064 Log stalled splitter outputs

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-5.6.8' into candidate-6.0.6

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.x' into candidate-6.0.6

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.6'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

jakesmith committed:
HPCC-16193 Prevent lightweight selfjoin deadlocking on stop

Lightweight selfjoin was calling stop twice, which could deadlock
some subgraphs

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

ghalliday committed:
HPCC-8603 When assign to SELF.x.y also check if SELF.x assigned

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

richardkchapman committed:
Merge pull request #9049 from GordonSmith/HPCC-15848

HPCC-15848 Graph properties fail to display

Reviewed-By: Miguel Vazquez <miguel.vazquez@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8959 from garonsky/HPCC-15962_add_option_to_specify_multiple_esps

HPCC-15962 Envgen - add options to specify multiple esp instances

Reviewed-By: Michael Gardner <michael.gardner@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9050 from ghalliday/issue16142

HPCC-16142 Fix problems creating spills from dictionaries

Reviewed-By: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-5.6.8' into candidate-6.0.6

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.6'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9036 from jakesmith/hpcc-16121

HPCC-16121 Problems with virtual logicalfilename in superfiles

Reviewed-By: Mark Kelly <mark.kelly@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #9024 from richardkchapman/thread-priority

HPCC-16117 UDP messaging threads should run at elevated priority

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #9041 from mckellyln/hpcc-16016b

HPCC-16016 CheckAbnormalTermination could incorrectly report job failed

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #9054 from jakesmith/hpcc-16193

HPCC-16193 Prevent lightweight selfjoin deadlocking on stop

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-5.6.8' into candidate-6.0.6

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-5.6.8' into candidate-6.0.6

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Community Edition 6.0.6-rc2 Release Candidate 2

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

mckellyln committed:
HPCC-16200 Systemd hpcc startup file perms cause kernel msgs

Signed-off-by: Mark Kelly <mark.kelly@lexisnexis.com>

richardkchapman committed:
Merge branch 'candidate-5.6.8' into candidate-6.0.6

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Community Edition 6.0.6-rc2 Release Candidate 2 (retag)

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9057 from Michael-Gardner/HPCC-16194

HPCC-16194 SocketEndpointv4 added getUrlStr(StringBuffer &) to struct

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #9035 from richardkchapman/ok-to-send

HPCC-16130 Change default timeout on ok_to_send response

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

mckellyln committed:
HPCC-16200 Systemd hpcc startup file perms cause kernel msgs

Signed-off-by: Mark Kelly <mark.kelly@lexisnexis.com>

richardkchapman committed:
Merge pull request #9052 from miguelvazq/HPCC-16191

HPCC-16191 Add missing alerts in ECL Watch

Reviewed-By: Gordon Smith <gordon.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

jakesmith committed:
HPCC-16051 A couple of problems in prev. commit spotted/fixed

NB: This is mostly tidying up, removing a couple of unused methods
and unecssary member variables (involve onStart())

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

ghalliday committed:
Merge pull request #9027 from jakesmith/hpcc-16051.master

HPCC-16051 Fix problems with child act. initialization

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

rpastrana committed:
HPCC-16162 Version intereface chages

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

richardkchapman committed:
Merge branch 'candidate-5.6.8' into candidate-6.0.6

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-16213 - Thor allocates 0 pages on OSX

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9047 from rpastrana/HPCC-16162-ProvideESDLServiceName

HPCC-16162 Parse out ESDLService name for UI consumption

Reviewed-By: Kevin Wang <kevin.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-16140 Roxie fails with assert(factory) error if workunit submitted multiple times

Reduce (massively) the chances of unexpected hash collisions between
cloned workunits submitted close together.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #9067 from richardkchapman/osx_memsize

HPCC-16213 - Thor allocates 0 pages on OSX

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #9068 from richardkchapman/roxie-query-hash

HPCC-16140 Roxie fails with assert(factory) error if workunit submitted multiple times

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
HPCC-16012 Fix class name clash between Thor and Roxie

This could cause issues when unit testing if both versions get loaded.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

jakesmith committed:
HPCC-16132 Missing foreign subfiles cause problems to superfile ops.

If a foreign subfile is deleted, the superfile removed the entry
(for the purposes of ignoring it) from the meta info temporarilty
whilst the superfile was in use.
Unfortunately that caused problems if the use context was
manipulating the superfile, e.g. adding/removing subfiles.

It also prevented that missing superfile being removed from the
superfile for the same reason.

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

richardkchapman committed:
HPCC-16014 Contributors file needs some refreshing

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.6'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8960 from garonsky/HPCC-15637_specify_thor_slaves_and_master

HPCC-15637 envgen - set thor master option

Reviewed-By: Michael Gardner <michael.gardner@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8985 from ghalliday/issue16018

HPCC-16018 Implement blocked heap allocators

Reviewed-By: Mark Kelly <mark.kelly@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9007 from Michael-Gardner/hpcc-16098

HPCC-16098 Cmake configure target now sets up ssh keys if not present

Reviewed-By: Xiaoming Wang <xiaoming.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9064 from mckellyln/systemd_perms

Systemd hpcc startup file perms cause kernel msgs

Reviewed-By: Michael Gardner <michael.gardner@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9048 from rpastrana/HPCC-16181-DESDL-ConfigPassThrough

HPCC-16181 Mechanism for pass-through method configuration

Reviewed-By: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

jakesmith committed:
HPCC-16223 Fix CMPChannel object leak

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

ghalliday committed:
Merge pull request #9044 from jakesmith/hpcc-10132

HPCC-10132 Implement constant rhs for lookup/smart join

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

mckellyln committed:
HPCC-16227 Configmgr crash generating new xml

Signed-off-by: Mark Kelly <mark.kelly@lexisnexis.com>

rpastrana committed:
HPCC-16226 Echo resulting binding

- Provides option to echo binding upon publish, or configuration change
- Provides option to echo defintion upon publish

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

richardkchapman committed:
Merge pull request #9076 from mckellyln/configmgr_fix

HPCC-16227 Configmgr crash generating new xml

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9034 from Michael-Gardner/HPCC-16124

HPCC-16124 GPG Signing changes

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #9074 from jakesmith/hpcc-16223

HPCC-16223 Fix CMPChannel object leak

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #9071 from richardkchapman/contribs

HPCC-16014 Contributors file needs some refreshing

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #9070 from jakesmith/hpcc-16132

HPCC-16132 Missing foreign subfiles cause problems to superfile ops.

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #9069 from richardkchapman/roxie-lazyfileio

HPCC-16012 Fix class name clash between Thor and Roxie

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #9066 from jakesmith/hpcc-16051b

HPCC-16051 A couple of problems in prev. commit spotted/fixed

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

rpastrana committed:
HPCC-16226 Code review changes

- Do not attempt to echo new changes, if changes failed
- Do no setown when querying  a tree

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

rpastrana committed:
HPCC-16249 All string results were inexplicably nulled out

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

rpastrana committed:
HPCC-16251 Improve Ptree memory management

- Adding a ptree to another ptree was causing incorrect pointer counter

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

rpastrana committed:
HPCC-16249 Provide const static NULL Field processor

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

afishbeck committed:
HPCC-9230 remove unused php ESP sub_service defines

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

afishbeck committed:
HPCC-13714 Option for #Webservice(Fields()) to include unspecified fields

The following syntax:
#WEBSERVICE(FIELDS('field1','field2','*'));

The addition of the name '*' would cause the form to show
first field1 and field2 and then all additional stored inputs
(unless noinput is specified for that stored).

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

afishbeck committed:
HPCC-9440 Support HTTP Expect: 100-continue in Roxie and ESP

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

richardkchapman committed:
Merge pull request #9081 from rpastrana/HPCC-16251PTreeMemMan

HPCC-16251 Improve Ptree memory management

Reviewed-By: Kevin Wang <kevin.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9060 from ghalliday/issue8603

HPCC-8603 When assign to SELF.x.y also check if SELF.x assigned

Reviewed-By: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9084 from afishbeck/expect100continue

HPCC-9440 Support HTTP Expect: 100-continue in Roxie and ESP

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9082 from afishbeck/phpRemnant

HPCC-9230 remove unused php ESP sub_service defines

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

jakesmith committed:
HPCC-16254 Fix regression in child group master activity.

Regression caused by changes in HPCC-16051 caused a child group
activity in thormaster to crash during stats deserialization.

Move stats helper objects to the constructor.

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

richardkchapman committed:
Merge pull request #9085 from jakesmith/hpcc-16254

HPCC-16254 Fix regression in child group master activity.

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

jakesmith committed:
HPCC-16147 Fix keyedjoin skip bug.

A SKIP in a keyed join transform would skip all RHS matches in
the right group.

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

jakesmith commented:
@richardkchapman - please review

HPCCSmoketest commented:
Automated Smoketest 
Sha: 2274c6d466687f7d04f5964d5d40c8204802b8cd
Build: success
Install hpccsystems-platform-community-with-plugins_5.6.8-rc1.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK

wangkx committed:
HPCC-16260 Support new public parameter per DZ

Read @ECLWatchVisible from a DropZone section of environment.xml.
If it is set to false, the dropzone will not be retrieved by
the following calls used by ECLWatch.

the calls to display spray/despray pages.
FileSpray.DropZoneFiles for uploading files.
WsTopology.TpServiceQuery for server list page.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

rpastrana committed:
HPCC-16264 Couchbase plugin accept port for connection string

- Accept port and use it in connection string
- Default port to actual default port 8091
- Cleanup error report.

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

rpastrana committed:
HPCC-16257 Provide Binding node to sec manager plugins

- Provide binding subtree to plugins
- Rename parameter to reflect new tree content
- Add sample use in HTPass plugin (template security plugin)

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

richardkchapman committed:
Merge pull request #9080 from rpastrana/HPCC-16249NulledStrings

HPCC-16249 Couchbaseembedded - All string results were inexplicably nulled out

Reviewed-By: Russ Whitehead <william.whitehead@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9077 from rpastrana/HPCC-16226-EchoBindingTree

HPCC-16226 WsDESDLConfig methods should echo resulting binding/def

Reviewed-By: Kevin Wang <kevin.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9083 from afishbeck/storedWebserviceAdditionalFields

HPCC-13714 Option for #Webservice(Fields()) to include unspecified fields

Reviewed-By: Kevin Wang <kevin.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9092 from wangkx/h16260

HPCC-16260 Support new public parameter per DZ

Reviewed-By: Mark Kelly <mark.kelly@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

rpastrana committed:
HPCC-16264 Code review change

- Minor removed superfluous conditional

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

rpastrana committed:
HPCC-16257 Code review changes

- Renamed input param
- Encapsulated Owned objects for quick release

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

richardkchapman committed:
Merge pull request #9094 from rpastrana/HPCC-16264-Couchbase-AcceptPort

HPCC-16264 Couchbase plugin accept port for connection string

Reviewed-By: Russ Whitehead <william.whitehead@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9095 from rpastrana/HPCC-16257-ProvideBindingNode

HPCC-16257 SecManager should provide binding node to plugins

Reviewed-By: Russ Whitehead <william.whitehead@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

rpastrana committed:
HPCC-15541 ESDL Publish binding usage update

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

RussWhitehead committed:
HPCC-16105 Fileservices checkExternalFileRights() incorrect check

In fileservices.cpp, method checkExternalFileRights accepts 2 flags indicating
it should test for read, and one for write. The "if" statement presumes that
only one of these can be true. Additionally, the "extpath" variable is never
set, so the code never checks the specified file. This PR sets the path
variable correctly, and checks for both read and write perms independently.

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

wangkx committed:
HPCC-15791 Add ECLWatchVisible to ConfigMgr

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

RussWhitehead committed:
Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

richardkchapman committed:
Merge pull request #9103 from wangkx/h15791

HPCC-15791 Add ECLWatchVisible to ConfigMgr

Reviewed-By: Mark Kelly <mark.kelly@lexisnexis.com>
Reviewed-By: Attila Vamos <attila.vamos@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9101 from rpastrana/HPCC-15541UpdatePublishDefinitionUsage

HPCC-15541 ESDL Publish binding usage update

Reviewed-By: Kevin Wang <kevin.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #9102 from RussWhitehead/checkLFN

HPCC-16105 Fileservices checkExternalFileRights() incorrect check

Reviewed-By: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

jakesmith commented:
@richardkchapman - This PR is targeted at 5.6.8
The fix is fairly trivial - but as you say (in the JIRA) it changes behaviour - so probably should be in a point build at all.
Shall I retarget to 6.2 ?

jakesmith commented:
Closing to retarget
